modules/objdetect/src/qrcode.cpp

```diff
@@ -1245,5 +1228,1206 @@ cv::String QRCodeDetector::detectAndDecode(InputArray in,
+      if(contours.size() < 1) return false;
+      int count = 0;
+      double eps = 3;
+      for(size_t i = 0; i < all_localization_points.size(); i++)
+      {
+            double dist;
+            if((dist = pointPolygonTest(contours[0], all_localization_points[i], false)) > 0)
+            {
+
+               if((abs(all_localization_points[i].x - p0.x) > eps) && (abs(all_localization_points[i].x - p1.x) > eps)
+               && (abs(all_localization_points[i].x - p2.x) > eps))
+               {
+                  return false;
+               }
+            }
+       }
+       if (count == 0) return true;
+       else return false;
+}
+
+bool QRDetectMulti::compareSquare::operator()(const Vec3i& a, const Vec3i& b) const
+{
+  return abs((points[a[1]].x - points[a[0]].x) *
+  (points[a[2]].y - points[a[0]].y) -
+  (points[a[2]].x - points[a[0]].x) *
+  (points[a[1]].y - points[a[0]].y)) <
+  abs((points[b[1]].x - points[b[0]].x) *
+  (points[b[2]].y - points[b[0]].y) -
+  (points[b[2]].x - points[b[0]].x) *
+  (points[b[1]].y - points[b[0]].y));
+}
+
+bool QRDetectMulti::localization()
+{
+   CV_TRACE_FUNCTION();
+   vector<Point2f> tmp_localization_points;
+   int num_qrcodes;
+   size_t number_possible_purpose = 1;
+   if(purpose == SHRINKING) number_possible_purpose = 2;
+   Mat tmp_shrinking = bin_barcode;
+   int tmp_num_points = 0;
+   int num_points = -1;
+   for(eps_horizontal = 0.1; eps_horizontal < 0.4; eps_horizontal += 0.1)
+   {
+       tmp_num_points = 0;
+       num_points = -1;
+       if(purpose == SHRINKING) number_possible_purpose = 2;
+       else number_possible_purpose = 1;
+       for(size_t k = 0; k < number_possible_purpose; k++)
+       {
+           if(k == 1) bin_barcode = bin_barcode_fullsize;
+           vector<Vec3d> list_lines_x = searchHorizontalLines();
+           if( list_lines_x.empty() )
+           {
+               if(k == 0)
+               {
+                   k = 1;
+                   bin_barcode = bin_barcode_fullsize;
+                   list_lines_x = searchHorizontalLines();
+                   if (list_lines_x.empty()) break;
+               }
+               else break;
+           }
+           vector<Point2f> list_lines_y = separateVerticalLines(list_lines_x);
+           if ( list_lines_y.size() < 3 )
+           {
+               if(k == 0)
+               {
+                   k = 1;
+                   bin_barcode = bin_barcode_fullsize;
+                   list_lines_x = searchHorizontalLines();
+                   if (list_lines_x.empty()) break;
+                   list_lines_y = separateVerticalLines(list_lines_x);
+                   if( list_lines_y.size() < 3 ) break;
+               }
+               else break;
+            }
+            vector<int> index_list_lines_y;
+            for(size_t i = 0; i < list_lines_y.size(); i++)
+                index_list_lines_y.push_back(-1);
+            num_points = 0;
+            for(size_t i = 0; i < list_lines_y.size() - 1; i++)
```

Comment:
Consider adding high-level non-obvious comments into your code.
What and why are these loops doing here? What are main stages of this algorithm?

This method is ~400 lines. It definitely should be divided into well named (and well tested) parts/stages.